ibis/pandas/dispatch.py

```diff
@@ -1,15 +1,36 @@
 from __future__ import absolute_import
 
-from functools import partial
+from functools import partial, singledispatch
 
 import toolz
 from multipledispatch import Dispatcher
 
 import ibis
 import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
 import ibis.expr.operations as ops
+import ibis.expr.types as ir
+import ibis.expr.window as win
 from ibis.pandas.trace import TraceTwoLevelDispatcher
 
+
+@singledispatch
+def is_computable_input(arg):
```

Comment:
Can we move these back to core.py? I don't feel the need to move these under dipatch.py. Dispatch.py contains things that are more generic.

Reply:
OK I will put it back to core.py and move new compute_time_context to a new file.

Reply:
There might be some circle deps issues I see. I don't think you can move compute_time_context to a new file. For now I think it's fine to keep it in core.py?

Reply:
Decided to keep it in core.py.